src/cloud-api-adaptor/entrypoint.sh

```diff
@@ -85,6 +85,22 @@ azure() {
         ${optionals}
 }
 
+gcp() {
+test_vars GCP_CREDENTIALS
+
+[[ "${PODVM_IMAGE_NAME}" ]] && optionals+="-gcp-image-name ${PODVM_IMAGE_NAME} "
+[[ "${GCP_PROJECT_ID}" ]] && optionals+="-gcp-project-id ${GCP_PROJECT_ID} "
+[[ "${GCP_ZONE}" ]] && optionals+="-gcp-zone ${GCP_ZONE} " # if not set retrieved from IMDS
```

Comment:
Does GCP has an equivalent to AWS's IMDS?

AWS implements https://github.com/confidential-containers/cloud-api-adaptor/issues/315 which will get information like the subnetId from the IMDS. This means that the subnetId parameter is optional.

Is GCP_PROJECT_ID and GCP_ZONE automatically filled if the user doesn't set? If not, then these variables should be listed in `test_vars` because they become mandatory.

`PODVM_IMAGE_NAME` should be mandatory, thus in `test_vars`.

Reply:
It does, and we are using in other parts, like: https://github.com/beraldoleal/cloud-api-adaptor/blob/d7578e6beb1e884be63888036a2748ef6347c857/src/cloud-providers/gcp/misc.go#L47.

This is just a basic implementation, for tech-preview, which will have lots of rooms for improvements. I will note this one as an issue once this gets merged, so we can improve latter. 

About the variables, you are right, sending the fix with the next force-push.